realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -173,9 +160,25 @@ private void emitAccessors(JavaWriter writer) throws IOException {
                 writer.emitStatement(
                         "realm.checkIfValid()"
                 );
+
+                // For String and bytes[], null value will be returned by JNI code. Try to save one JNI call here.
+                if (metadata.isNullable(field) && !Utils.isString(field) && !Utils.isByteArray(field)) {
+                    writer.beginControlFlow("if (row.isNull(%s))", staticFieldIndexVarName(field));
+                    writer.emitStatement("return null");
+                    writer.endControlFlow();
+                }
+
+                // For Boxed types, this would be the corresponding primitive types. Others remain the same.
+                String castingBackType = fieldTypeCanonicalName;
+                try {
+                    Types typeUtils = processingEnvironment.getTypeUtils();
+                    castingBackType = typeUtils.unboxedType(field.asType()).toString();
+                } catch (IllegalArgumentException ignored) {
+                    // Not a boxed type, do nothing then.
```

Comment:
Using exceptions to control flow is bad (tm).  Isn't there a better way around this?